pkg/spdx/document.go

```diff
@@ -0,0 +1,254 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package spdx
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	spdx_json "github.com/spdx/tools-golang/json"
+
+	"github.com/spdx/tools-golang/spdx/common"
+	"github.com/spdx/tools-golang/spdx/v2_2"
+	"github.com/spdx/tools-golang/spdx/v2_3"
+)
+
+const errOpenDoc = "opening SPDX %s document: %w"
+
+var ErrUnknownFormat = fmt.Errorf("unrecognized document format")
+
+type Document_22 v2_2.Document
+type Document_23 v2_3.Document
+
+type Package struct {
+	PackageLicenseConcluded   string
+	PackageExternalReferences []*PackageExternalReference
+	PackageChecksums          []common.Checksum
+	PackageVersion            string
+}
+
+type File struct {
+	Checksums []common.Checksum
+}
+
+func LoadDocument(path string) (Document, error) {
+	var err1 error
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("opening SPDX document: %w", err)
+	}
+
+	doc23, err1 := spdx_json.Load2_3(f)
+	if err1 == nil && doc23 != nil {
+		return documentFromSPDX(doc23)
+	}
+
+	// First, try to open SPDX 2.2
+	doc22, err := spdx_json.Load2_2(f)
+	if err != nil {
+		err := fmt.Errorf(errOpenDoc, "v2.2", err)
+		if err1 != nil {
```

Comment:
Just curious, why try to load in v2.3 on line 42, but handle error messages for it down here? Ideally the logic would be put near where the thing happens.